### PR TITLE
[DCCsi] Maya Scene Materials Exporter

### PR DESCRIPTION
**Pull request by HogJonny-AMZN**
## What does this PR do?

This PR integrates into the latest development a Maya Material Exporter/Conversion tool written by @benblack75 
Ben's commits were cherry-picked from his branch into this one, then some refactoring occurred to get it running with the latest changes in development.

original branch PR I cherry-picked over: https://github.com/o3de/o3de/pull/11634

## How was this PR tested?

You can start Maya with the developer windows .bat file environment:
"C:\path\to\o3de\Gems\AtomLyIntegration\TechnicalArt\DccScriptingInterface\Tools\Dev\Windows\DCC\Launch_Maya_2022.bat"

1. launch maya, check boot console output window, and then script editor console to verify DCCsi boo strapping
2. Verified that a o3de:dccsi menu exists in the Maya menu bar
3. Verified that the Material Helper utility can be started from the menu's
4. Attempted to convert a stingraypbs material (currently broken as there is no standardpbr_allproperties.material template being generated for this tool to load.)

Additional info:
Here is a screenshot of the tool:
maya_dFeKjzIkJL

This bug fix is included: https://github.com/o3de/o3de/issues/12181

This tool works if I copy an old standardpbr_allproperties.material template file into my project cache

### CODE REVIEW

**Review by gadams3 — APPROVED** [review]
I'm approving based on a cursory review and sanity checks. I will need time and demonstration of all the DCCsi Python tools to do an in-depth review

**Comment by santorac** [comment]
It would be helpful to see some screenshots of the tool.

**Review by santorac — APPROVED** [review]
I'm not experience enough with this system to give a deep review, but I don't see anything I would consider blocking.

**Comment by HogJonny-AMZN** [comment]
I need to include fixes for this GHI I encountered on a clean machine:
https://github.com/o3de/o3de/issues/12181

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/azpy/dcc/maya/helpers/maya_materials_conversion.py [hunk 1]
```diff
@@ -0,0 +1,508 @@
...
+
+def get_maya_file_path(fbx_file_path):
+    """! Convenience function for creating a maya file path based on an existing FBX path.
+    @param fbx_file_path Path to the existing FBX file
+    @return str - File path
+    """
+    return os.path.splitext(fbx_file_path)[0] + '.ma'
+
+
+def get_material(material_name: str) -> list:
```
[the remark this answers is not in the document]
Comment (HogJonny-AMZN) [reply Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/azpy/dcc/maya/helpers/maya_materials_conversion.py:299]: Logged concern in a GHI as a request: https://github.com/o3de/o3de/issues/12415

**Pull request merged**